### DISCUSSION TITLE
Flex diskdefs

### DISCUSSION THREAD

**Discussion by Wawavoun** [keirf/greaseweazle#278] [General]
Hello there !

Is it possible to add to diskdefs.cfg something for write image with this definition (take from Flashfloppy IMG.CFG) ?

[::733184]   80 Tracks  dsdd
cyls = 80
heads = 2
bps = 256       # All tracks have 256-byte sectors. Number of sectors varies.
iam = no
gap3 = 16
gap4a = 16
tracks = 0.0    ## Boot cylinder, head 0
  secs = 10
  mode = fm
  interleave = 4
  id = 1
  h = 0
tracks = 0.1    ## Boot cylinder, head 1
  secs = 10
  mode = fm
  interleave = 4
  hskew = 1     # Simulates correct inter-track interleave
  id = 11       # Follows on from head 0 (1..10 -> 11..20)
  h = 0
tracks = 1-79.0 ## All other cylinders, head 0
  secs = 18
  mode = mfm
  interleave = 6
  id = 1
  h = 1
tracks = 1-79.1 ## All other cylinders, head 1
  secs = 18
  mode = mfm
  interleave = 6
  hskew = 5     # Simulates correct inter-track interleave
  id = 19       # Follows on from head 0 (1..18 -> 19..36)
  h = 1

It is a TCS Flex floppy definition. There are other ones (see joined file) but should be easy to do after this one... 

Thanks and regards.
Philippe

[IMG.CFG.zip](https://github.com/keirf/greaseweazle/files/10610641/IMG.CFG.zip)

**Comment by keirf** [comment]
Here is both Flex configs converted for Greaseweazle. Unzip and use something along the lines of these example command lines (one for reading from disk, another converting an existing raw SCP flux file). If you can confirm they work correctly, I will add them to the default diskdefs config file for the next Greaseweazle release!

```
gw read --diskdefs=flex_diskdefs.cfg --format=tsc.flex.dsdd my.img
gw convert --diskdefs=flex_diskdefs.cfg --format=tsc.flex.dsdd my.scp my.img
```

[flex_diskdefs.zip](https://github.com/keirf/greaseweazle/files/10610854/flex_diskdefs.zip)

**Reply by Wawavoun** [reply]
Thanks a lot Keir !
I will keep you informed as soon I can test it.

**Comment by Wawavoun** [comment]
Hello,

At least DSDD description works well, for reading and writing Flex images. The floppies can boot perfectly and are usable.
I have use real DD floppies and also HD floppies with the HD hole covered.
I have not test the SSDD description but should also works.

Thanks Keir.

Philippe

**Reply by keirf** [reply]
Thanks. I have added the diskdefs to the Greaseweazle tree to be included in the next tools release.
